### ISSUE TITLE
fixed_regex_linter() throws false positives for piped-in `stringr` calls

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1811]
```r
x %>%
  str_replace(fixed("abc"), "cde")
```

Since our logic currently uses the argument number and doesn't account for whether the call is piped-in, it sees `cde` as being "fixed" and hence throws a lint.
